smartsim/experiment.py

```diff
@@ -136,6 +163,42 @@ def __init__(
         self._launcher = launcher.lower()
         self.db_identifiers: t.Set[str] = set()
 
+        contextualize(self, self.start, ctx_exp_path)
+        contextualize(self, self.stop, ctx_exp_path)
+        contextualize(self, self.generate, ctx_exp_path)
+        contextualize(self, self.poll, ctx_exp_path)
+        contextualize(self, self.finished, ctx_exp_path)
+        contextualize(self, self.get_status, ctx_exp_path)
+
+    @property
+    def context_value(self) -> str:
+        return self.exp_path
+
+    def _contextual(self, func: t.Callable[..., t.Any]) -> t.Callable[..., t.Any]:
```

Comment:
not used in this example, just an "inside version" of the contextualize function outside Experiment